pkg/cmd/cli/describe/deployments.go

```diff
@@ -28,99 +29,25 @@ import (
-func NewDeploymentConfigDescriberForConfig(client client.Interface, kclient kclient.Interface, config *deployapi.DeploymentConfig) *DeploymentConfigDescriber {
-	return &DeploymentConfigDescriber{
-		client: &genericDeploymentDescriberClient{
-			getDeploymentConfigFunc: func(namespace, name string) (*deployapi.DeploymentConfig, error) {
-				return config, nil
-			},
-			getDeploymentFunc: func(namespace, name string) (*kapi.ReplicationController, error) {
-				return nil, kerrors.NewNotFound(kapi.Resource("replicatoncontroller"), name)
-			},
-			listDeploymentsFunc: func(namespace string, selector labels.Selector) (*kapi.ReplicationControllerList, error) {
-				return nil, kerrors.NewNotFound(kapi.Resource("replicationcontrollerlist"), fmt.Sprintf("%v", selector))
-			},
-			listPodsFunc: func(namespace string, selector labels.Selector) (*kapi.PodList, error) {
-				return nil, kerrors.NewNotFound(kapi.Resource("podlist"), fmt.Sprintf("%v", selector))
-			},
-			listEventsFunc: func(deploymentConfig *deployapi.DeploymentConfig) (*kapi.EventList, error) {
-				return kclient.Events(config.Namespace).Search(config)
-			},
-		},
-	}
+	osClient   client.Interface
+	kubeClient kclient.Interface
 }
 
 // NewDeploymentConfigDescriber returns a new DeploymentConfigDescriber
 func NewDeploymentConfigDescriber(client client.Interface, kclient kclient.Interface) *DeploymentConfigDescriber {
 	return &DeploymentConfigDescriber{
-		client: &genericDeploymentDescriberClient{
-			getDeploymentConfigFunc: func(namespace, name string) (*deployapi.DeploymentConfig, error) {
-				return client.DeploymentConfigs(namespace).Get(name)
-			},
-			getDeploymentFunc: func(namespace, name string) (*kapi.ReplicationController, error) {
-				return kclient.ReplicationControllers(namespace).Get(name)
-			},
-			listDeploymentsFunc: func(namespace string, selector labels.Selector) (*kapi.ReplicationControllerList, error) {
-				return kclient.ReplicationControllers(namespace).List(kapi.ListOptions{LabelSelector: selector})
-			},
-			listPodsFunc: func(namespace string, selector labels.Selector) (*kapi.PodList, error) {
-				return kclient.Pods(namespace).List(kapi.ListOptions{LabelSelector: selector})
-			},
-			listEventsFunc: func(deploymentConfig *deployapi.DeploymentConfig) (*kapi.EventList, error) {
-				return kclient.Events(deploymentConfig.Namespace).Search(deploymentConfig)
-			},
-		},
+		osClient:   client,
+		kubeClient: kclient,
 	}
 }
 
 // Describe returns the description of a DeploymentConfig
 func (d *DeploymentConfigDescriber) Describe(namespace, name string) (string, error) {
-	deploymentConfig, err := d.client.getDeploymentConfig(namespace, name)
+	deploymentConfig, err := d.osClient.DeploymentConfigs(namespace).Get(name)
 	if err != nil {
 		return "", err
 	}
-	events, err := d.client.listEvents(deploymentConfig)
+	events, err := d.kubeClient.Events(deploymentConfig.Namespace).Search(deploymentConfig)
 	if err != nil {
 		return "", err
```

Comment:
This should not have ever failed on this.  Fix it while you're here.  An error on secondary retrieval should indicate that the lookup failed, but not crater the describe